02-starting-with-data.Rmd

```diff
@@ -111,10 +111,143 @@ Based on the output of `str(surveys)`, can you answer the following questions?
+          and the number of columns as the second element (the **dim**ensions of
+          the object)
+    * `nrow(surveys)` - returns the number of rows
+    * `ncol(surveys)` - returns the number of columns
+
+* Content:
+    * `head(surveys)` - shows the first 6 rows
+    * `tail(surveys)` - shows the last 6 rows
+
+* Names:
+    * `names(surveys)` - returns the column names (synonym of `colnames()` for `data.frame`
+	   objects)
+    * `rownames(surveys)` - returns the row names
+
+* Summary:
+    * `str(surveys)` - structure of the object and information about the class, length and
+	   content of  each column
+    * `summary(surveys)` - summary statistics for each column
+
+Note: most of these functions are "generic", they can be used on other types of
+objects besides `data.frame`.
+
+## Indexing and subsetting data frames
+
+```{r, echo=FALSE, purl=TRUE}
+
+## Sequences and Subsetting data frames
+```
+
+
+Our survey data frame has rows and columns (it has 2 dimensions), if we want to
+extract some specific data from it, we need to specify the "coordinates" we
+want from it. Row numbers come first, followed by column numbers. However, note
+that different ways of specifying these coordinates lead to results with
+different classes.
+
+```{r, purl=FALSE}
+surveys[1]      # first column in the data frame (as a data.frame)
+surveys[, 1]     # first column in the data frame (as a vector)
+surveys[1, 1]   # first element in the first column of the data frame (as a vector)
+surveys[1, 6]   # first element in the 6th column (as a vector)
+surveys[1:3, 7] # first three elements in the 7th column (as a vector)
+surveys[3, ]    # the 3rd element for all columns (as a data.frame)
+head_surveys <- surveys[1:6, ] # equivalent to head(surveys)
+```
+
+`:` is a special function that creates numeric vectors of integers in increasing
+or decreasing order, test `1:10` and `10:1` for instance.
+
+You can also exclude certain parts of a data frame using the "`-`" sign:
+
+```{r, purl=FALSE}
+surveys[,-1]   #The whole data frame, except the first column
+surveys[-c(7:34786),] #equivalent to head(surveys)
+```
+
+As well as using numeric values to subset a `data.frame` (or `matrix`), columns
+can be called by name, using one of the three following notations:
+
+```{r, eval = FALSE, purl=FALSE}
+surveys["species_id"]       # Result is a data.frame
+surveys[, "species_id"]     # Result is a vector
+surveys[["species_id"]]     # Result is a vector
+surveys$species_id          # Result is a vector
+```
+
+For our purposes, the last three notations are equivalent. However, the last one
+with the `$` does partial matching on the name. So you could also select the
+column `"day"` by typing `surveys$d`. It's a shortcut, as with all shortcuts,
+they can have dangerous consequences, and are best avoided. Besides, with
+auto-completion in RStudio, you rarely have to type more than a few characters
+to get the full and correct column name.
+
```

Comment:
This is probably more confusing than helpful. We probably don't need to introduce this way of selecting. As noted this strategy can lead to issues and is probably not good practice.